cpp/src/io/orc/writer_impl.cu

```diff
@@ -467,13 +469,8 @@ orc_streams writer::impl::create_streams(host_span<orc_column_view> columns,
     // Initialize the column's metadata (this is the only reason columns is in/out param)
     column.set_orc_encoding(encoding_kind);
 
-    // Initialize the column's index stream
-    const auto id      = static_cast<uint32_t>(1 + column.id());
-    streams[id].column = id;
-    streams[id].kind   = ROW_INDEX;
-    streams[id].length = 0;
-
     // Initialize the column's data stream(s)
+    const auto id   = static_cast<uint32_t>(1 + column.id());
```

Comment:
Recommend adding a function to the `orc_column_view` class, such as

`auto stream_id() const noexcept { return id() + 1; }`

Might also consider changing `id()` to `index()` since that appears to be how it's used. The semantics of "id" are a little hazy.

Reply:
Renamed to index and placed the index<->id logic in Stream class.